ast/desugar/Desugar.cc

```diff
@@ -758,13 +766,12 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                     if (numPosArgs > 0) {
                         // Deconstruct the kwargs hash in the last argument if it's present.
                         if (auto *hash = parser::cast_node<parser::Hash>(send->args.back().get())) {
-                            if (hash->kwargs) {
+                            if (hash->inline_) {
                                 numPosArgs--;
 
                                 // skip inlining the kwargs if there are any non-key/value pairs present
-                                if (absl::c_all_of(hash->pairs, [](auto &arg) {
-                                        return parser::isa_node<parser::Pair>(arg.get());
-                                    })) {
+                                if (absl::c_all_of(hash->pairs,
+                                                   [](auto &arg) { return isInlininableKwargPair(arg.get()); })) {
```

Comment:
This original condition was too lax: it was only looking for pairs present in the hash, not also that those pairs had keys that were symbols.